tooling/apt/src/main/java/org/apache/camel/tools/apt/CoreEipAnnotationProcessor.java

```diff
@@ -219,8 +219,8 @@ protected EipModel findEipModelProperties(ProcessingEnvironment processingEnv, R
         boolean deprecated = classElement.getAnnotation(Deprecated.class) != null;
         model.setDeprecated(deprecated);
 
-        Metadata metadata = classElement.getAnnotation(Metadata.class);
-        if (metadata != null) {
+        Metadata[] metadataArray = classElement.getAnnotationsByType(Metadata.class);
+        for (Metadata metadata : metadataArray) {
```

Comment:
I dont think we should just loop and override the label, eg we should get the metadata annotation that has no `key` set as that is what is used for components today. Or it should grab the first that is not empty.

Reply:
not sure what i was thinking. i will look into it again.

Reply:
> I dont think we should just loop and override the label, eg we should get the metadata annotation that has no `key` set as that is what is used for components today. Or it should grab the first that is not empty.

@davsclaus , i will try to get back to the old PR and rework on it. I have got a couple of questions.
As per your comment, 

1- below will result in that with only Metadata annotation with key="platform" will be taken care of and the other will be ignored.

`@Metadata(key = "platforms", enums = { "spring", "spring-boot", "osgi" })`
`@Metadata(types = { MyExtension.cass })`

2-  below will result in that with only Metadata annotation with key="platforms" will be taken care of and the other will be ignored.
`@Metadata(key = "platforms", enums = { "spring", "spring-boot", "osgi" })`
`@Metadata(key = "extensions", types = { MyExtension.cass })`

3- below will result in that with only Metadata annotation with enums will be taken care of and the other will be ignored.
`@Metadata(enums = { "spring", "spring-boot", "osgi" })`
`@Metadata(types = { MyExtension.cass })`

So it feels like using multiple Metadata annotation will make no difference.
Possibly i am getting something missing. 
Could you please help me understand?

@lburgazzoli , could you also comment in the JIRA issue about the purpose of use of multiple Metadata annotation and how it may affect the resulting json files?